app/src/marketplace/modules/streams/services.js

```diff
@@ -31,6 +31,40 @@ export const getStreams = (params: any): ApiResult<{
         }))
 }
 
+const STREAMS_PAGE_SIZE = 999
+
+export async function* getPagedStreams(params: Object): any {
+    let page = 0
+    let hasMore = false
+
+    do {
+        // eslint-disable-next-line no-await-in-loop
+        const { streams, hasMoreResults } = await getStreams({
+            ...(params || {}),
+            max: STREAMS_PAGE_SIZE,
+            offset: page * STREAMS_PAGE_SIZE,
+        })
+        page += 1
+        hasMore = hasMoreResults
+
+        yield streams
+    } while (hasMore)
+}
+
+export async function getAllStreams(params: Object): any {
+    let streams = []
+
+    // eslint-disable-next-line no-restricted-syntax, no-await-in-loop
+    for await (const pagedStreams of getPagedStreams(params)) {
+        streams = [
+            ...streams,
+            ...pagedStreams,
+        ]
```

Comment:
Could this be `streams.push(...pagesStreams)`? `streams` would be a constant in that case.